src/rendering/src/components/Navigation/HeaderContent.tsx

```diff
@@ -29,23 +29,23 @@ const HeaderContent = (props: HeaderContentProps): JSX.Element => {
           query: props.query,
         },
         props.asPath,
-        { locale: lang }
+        {
+          locale: lang,
+          shallow: false,
+        }
       );
     }
   };
 
   const languageSelector = languageList && (
-    <select
-      onChange={(e) => changeLanguage(e.currentTarget.value)}
-      className="languagePicker"
-      defaultValue={sitecoreContext.language}
```

Comment:
I tested and I think `value={sitecoreContext.language}` would work instead of `defaultValue={sitecoreContext.language}`

Reply:
Thanks, updated it and it works well!